xclim/indices/_agro.py

```diff
@@ -506,6 +525,199 @@ def cool_night_index(
+        If None, a CF-conformant "latitude" field must be available within the passed DataArray.
+    wo : str
+        The initial soil water reserve accessible to root systems [length]. Default: 200 mm.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [mm]
+        Dryness Index.
+
+    Notes
+    -----
+    Given that this indice only examines monthly total accumulations for six-month periods depending on the hemisphere,
+    it is possible to send in DataArrays containing only these timesteps. Users should be aware that due to the missing
+    values checks in wrapped Indicators, datasets that are missing several months will be flagged as invalid. This check
+    can be ignored by setting the following context:
+
+    .. code-block:: python
+
+        with xclim.set_options(check_missing="skip"):
+            di = dryness_index(pr, evspsblpot)
+
+    Let :math:`Wo` be the initial useful soil water reserve (typically "200 mm"), :math:`P` be precipitation,
+    :math:`T_{v}` be the potential transpiration in the vineyard, and :math:`E_{s}` be the direct evaporation from the
+    soil. Then the Dryness Index, or the estimate of soil water reserve at the end of a period (1 April to 30 September
+    in the Northern Hemispherere or 1 October to 31 March in the Southern Hemisphere), can be given by the following
+    formulae:
+
+    .. math::
+
+        W = \sum_{\text{April 1}}^{\text{September 30}} \left( Wo + P - T_{v} - E_{s} \right)
+
+    or (for the Southern Hemisphere):
+
+    .. math::
+
+        W = \sum_{\text{October 1}}^{\text{March 31}} \left( Wo + P - T_{v} - E_{s} \right)
+
+    Where :math:`T_{v}` and :math:`E_{s}` are given by the following formulae:
+
+    .. math::
+
+        T_{v} = ETP * k
+
+    and
+
+    .. math::
+
+        E_{s} = \frac{ETP}{N}\left( 1 - k \right) * JPm
+
+    Where :math:`ETP` is evapotranspiration, :math:`N` is the number of days in the given month. :math:`k` is the
+    coefficient for radiative absorption given by the vine plant architecture, and :math:`JPm` is the number of days of
+    effective evaporation from the soil per month, both provided by the following formulae:
+
+    .. math::
+
+        k = \begin{cases}
+            0.1, & \text{if month = April (NH) or October (SH)}  \\
+            0.3, & \text{if month = May (NH) or November (SH)}  \\
+            0.5, & \text{if month = June - September (NH) or December - March (SH)} \\
+            \end{cases}
+
+    .. math::
+
+        JPm = \max\left( P / 5, N \right)
+
+    Examples
+    --------
+    >>> from xclim.indices import dryness_index
+    >>> dryi = dryness_index(pr_dataset, evspsblpot_dataset, wo="200 mm")
+
+    References
+    ----------
+    :cite:cts:`tonietto_multicriteria_2004,riou_determinisme_1994`
+
+    """
+    if parse_offset(freq) != (1, "A", True, "JAN"):
+        raise ValueError(f"Freq not allowed: {freq}. Must be `YS` or `AS-JAN`")
```

Comment:
`freq` is not used in the index apart from this line which just checks that is has a correct value, correct? Was there plans to allow a different resampling frequency that those below? If not, is there a reason to keep `freq` as an argument? It's to be used like other functions that do allow a greater flexibility for resampling?

Reply:
I think this has more to do with compatibility for `Indicators` that supply a `freq` argument than anything else. The value for this indicator is only valid for annual estimates, so I don't see any other values being allowed. There's already some handling for the Southern Hemisphere growing season, so that would provide even more reason to constrict this to `AS-JAN`.

I don't see any other `freq` values possible.